plugin/pkg/auth/authorizer/node/graph.go

```diff
@@ -233,6 +258,41 @@ func (g *Graph) removeVertex_locked(v *namedVertex) {
 	}
 }
 
+// must be called under write lock
+// recomputeDestinationIndex_locked recomputes the index of destination ids for the specified vertex
+func (g *Graph) recomputeDestinationIndex_locked(n graph.Node) {
+	// don't maintain indices for nodes with few edges
+	edgeCount := g.graph.Degree(n)
+	if edgeCount < g.destinationEdgeThreshold {
+		delete(g.destinationEdgeIndex, n.ID())
+		return
+	}
+
+	// get or create the index
+	index := g.destinationEdgeIndex[n.ID()]
```

Comment:
How much does not allocating a new index every recompute help?

Reply:
back of envelope calculations on a 5000 node cluster with high-scale replicasets (which is really the type of scenario this index is for):

* 60k per rebuild (5000 entries, average of 13 bytes per entry - https://play.golang.org/p/aTH-j0tSPCr)
* 2 highly shared objects per pod (service account and token secret)
* 2 rebuilds per object per graph-modifying object write (on edge removal and re-add)
* ~100 graph-modifying writes per second

That gives ~20MB/sec in allocations (~1.3GB/min, ~80GB/hour)

Reply:
added a benchmark to check index maintenance cost (time-wise and allocation-wise) processing a pod modification event on a 5000-node scenario:

with reuse (mark current members, sweep old members):
```
1508551 ns/op	    2034 B/op	      42 allocs/op
```

without reuse (reallocate every time we recompute the index):
```
1874243 ns/op	  271495 B/op	     282 allocs/op
```

so in this scenario, 265kb per op (probably from reallocation while growing the index from zero) vs 2kb for reuse.

interestingly, the reallocate case is also consistently slower than the mark/sweep case.